segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/SegmentAdminApi.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.contracts;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+
+public interface SegmentAdminApi {
+
+    /**
+     * Performs sanity operations on chunk like create chunk, write to the chunk, check if the chunk exists, read back contents to the chunk and delete the chunk.
+     *
+     * @param containerId The Id of the container on which sanity operations are to be performed.
+     * @param chunkName Name of the chunk on which sanity operations are to be performed.
+     * @param dataSize dataSize of the bytes to read.
+     * @param timeout Timeout for the operation.
+     * @return A Completable future that when completed, will indicate that the operation has been successfully completed.
+     */
+    default CompletableFuture<Void> checkChunkStorageSanity(int containerId, String chunkName, int dataSize, Duration timeout) {
+        throw new UnsupportedOperationException("checkChunkStorageSanity is not supported on " + getClass().getSimpleName());
```

Comment:
Looks like you'll have to add tests using mocks.

Reply:
Yes. The test coverage still doesn't seem to meet the requirement. Will work on adding some new tests using mock.